features/semantic-versioning.feature

```diff
@@ -0,0 +1,41 @@
+Feature: Calculating a version number based on semantic versioning
+
+  Scenario: Releasing a major version
+    Given a release on this branch with version "1.0.0"
+    When I release a new major version
+    Then version "2.0.0" should be released
+
+  Scenario: Releasing a minor version
+    Given a release on this branch with version "1.0.0"
+    When I release a new minor version
+    Then version "1.1.0" should be released
+
+  Scenario: Releasing a patch version
+    Given a release on this branch with version "1.0.9"
+    When I release a new patch version
+    Then version "1.0.10" should be released
```

Comment:
Not all cases (that seem common sense from a developers perspective) are present.
Does a major release from 1.0.1 really go to 2.0.0? Etc.

We might need some tests for the following cases.

* 1.0.1 > major > 2.0.0
* 1.1.0 > major > 2.0.0
* 1.1.1 > minor > 1.2.0
* 1.1.1 > patch > 1.1.2

And what happens when I release a patch or minor on 2.0.0-beta.2?